src/System.Memory/tests/MemoryMarshal/AsReadOnlyRef.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.SpanTests
+{
+    public static partial class MemoryMarshalTests
+    {
+        [Fact]
+        public static void AsReadOnlyRef()
+        {
+            ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(new byte[] { 0x11, 0x22, 0x22, 0x11 });
+            ref readonly int asInt = ref MemoryMarshal.AsRef<int>(span);
```

Comment:
How about adding a test case for interpreting a custom struct (both for success and failure, where the struct contains a reference type)?